### ISSUE TITLE
Create initial documentation website

### ISSUE THREAD

**Issue by shaido987** [huawei-noah/trustworthyAI#144]
There is no website with documentation available for gCastle so we want to add one. 

For similar open-source projects, using "read the docs" to host and sphinx to generate the documentation itself seems to be the most common approach. Something similar to https://libmtl.readthedocs.io/en/latest/ would be a great start.

**Comment by AlxndrMlk** [comment]
@shaido987 - so you'd like to use Sphinx to auto-generate the docs?

**Comment by shaido987** [comment]
@AlxndrMlk Yes. Not sure if it's doable with out current docstrings or not as none of us have any experience with sphinx. But should be worth a try to get a working foundation for a documentation website.

**Comment by AlxndrMlk** [comment]
@shaido987 - I'd require making the docstrings sphinx-compatible. Seems doable.

Happy to help adapting them.

**Comment by shaido987** [comment]
@AlxndrMlk Do you know if there are any larger changes required? The overall format should be ok using either the `numpydoc` or the `sphinxcontrib.napoleon` extension (not sure about the trade-offs here but both should provide support for numpy style docstrings). I just noticed that the dodiscover package has something similar (and uses `numpydoc`): https://github.com/py-why/dodiscover/blob/main/doc/conf.py

**Comment by AlxndrMlk** [comment]
It's pretty straightforward, @shaido987 

You can check this tutorial for more details, it's really good: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/index.html


PS: I didn't know about `numpydoc` - thanks for sharing.

**Comment by shaido987** [comment]
This has been completed with #172.
